src/backend/distributed/worker/worker_sql_task_protocol.c

```diff
@@ -251,7 +252,8 @@ TaskFileDestReceiverReceive(TupleTableSlot *slot, DestReceiver *dest)
 static void
 WriteToLocalFile(StringInfo copyData, File fileDesc)
 {
-	int bytesWritten = FileWrite(fileDesc, copyData->data, copyData->len, PG_WAIT_IO);
+	int bytesWritten = FileWriteCompat(fileDesc, copyData->data, copyData->len, 0,
```

Comment:
Passing `offset = 0` doesn't seem right here. In `TaskFileDestReceiverReceive` we might not be in the beginning of file.